lite/examples/sound_classification/ios/SoundClassification/TFLAudioRecord/TFLAudioRecord.m

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "TFLAudioRecord.h"
+#import "GMLRingBuffer.h"
+#import "GMLUtils.h"
+#import "GMLAudioError.h"
+
+#define SUPPORTED_CHANNEL_COUNT 1
+
+
+@implementation TFLAudioRecord {
+  AVAudioEngine *_audioEngine;
+  
+  /* Specifying a custom buffer size on AVAUdioEngine while tapping does not take effect. Hence we are storing the returned samples in a ring buffer to acheive the desired buffer size. If specified buffer size is shorter than the buffer size supported by AVAUdioEngine only the most recent data of the buffer of size bufferSize will be stored by the ring buffer. */
+  GMLRingBuffer *_ringBuffer;
+  dispatch_queue_t _delegateQueue;
+}
+
+-(nullable instancetype)initWithAudioFormat:(GMLAudioFormat *)audioFormat sampleCount:(NSUInteger)sampleCount error:(NSError *_Nullable *)error {
+  self = [self init];
+  if (self) {
+    if (audioFormat.channelCount > SUPPORTED_CHANNEL_COUNT) {
+      [GMLUtils createCustomError:error withCode:GMLAudioErrorCodeInvalidArgumentError description:@"The channel count provided does not match the supported channel count. Only 1 audio channel is currently supported."];
+      return nil;
+    }
+    _audioFormat = audioFormat;
+    _audioEngine = [[AVAudioEngine alloc] init];
+    _bufferSize = sampleCount * audioFormat.channelCount;
+    _ringBuffer = [[GMLRingBuffer alloc] initWithBufferSize:sampleCount * audioFormat.channelCount];
+    _delegateQueue = dispatch_queue_create("com.gmlAudio.AudioConversionQueue",
+                                           DISPATCH_QUEUE_CONCURRENT);
+  }
+  return self;
+}
+
+/**
+ * Uses AVAudioConverter to acheive the desired sample rate since the tap on the input node raises an exception if any format other than the input nodes default format is specified.
+ * completion handler structure is used as opposed to polling since this is the pattern followed by all iOS APIs delivering continuous data on a background thread.
+ * Even if we implement a read method, it will have to return data in a callback since installTapOnBus delivers data in a callback on a different thread. There will also be extra overhead to ensure thread safety to make sure that reads and writes happen on the same thread sine GMLAudio buffer is meant to be non local. 
```

Comment:
@khanhlvg Please see my note on why callback is used instead of polling like in Java.
Java AudioRecord read returns immediately as opposed to iOS installTapOnBus which delivers results in completion handler.